feat: [CDS-86606]: add best practices for expressions usage with method invocation (#4605)
harness/developer-hub: docs/platform/variables-and-expressions/harness-variables.md

   In the above expression, `<+pipeline.variables.var1>` must be wrapped in quotation marks because the expression is a string parameter for a method.

> - While using method invocation with an expression, the expression before method invocation should also be wrapped within <+...>

Reviewer: ```suggestion
- While using method invocation with an expression, the expression before method invocation should also be wrapped within `<+...>`.
```

  
  For example, consider the following expression:

  ```
  <+<+pipeline.variables.var1>.concat("concatenating a string")>
  ```
  

>   Here we want to invoke the method `concat` on the expression `<+pipeline.variables.var1>`, so, it is essential to wrap `<+pipeline.variables.var1>` within <+...> and then invoke method using `.concat()`

Reviewer: ```suggestion
  To invoke the method `concat` on the expression `<+pipeline.variables.var1>`, you must wrap `<+pipeline.variables.var1>` within `<+...>` and then invoke the method using `.concat()`.
```
